src/ring_theory/witt_vector/frobenius_fraction_field.lean

```diff
@@ -0,0 +1,587 @@
+import ring_theory.witt_vector.domain
+import ring_theory.witt_vector.truncated
+import data.mv_polynomial.supported
+
+
+/-!
+# Solving equations about the Frobenius map on the field of fractions of `𝕎 k`
+
+The goal of this file is to prove `witt_vector.exists_frobenius_solution_fraction_ring`,
+which says that for an algebraically closed field `k` of characteristic `p` and `a, b` in the
+field of fractions of Witt vectors over `k`,
+there is a solution `b` to the equation `φ b * a = p ^ m * b`, where `φ` is the Frobenius map.
+
+Most of this file builds up the equivalent theorem over `𝕎 k` directly,
+moving to the field of fractions at the end.
+See `witt_vector.frobenius_rotation` and its specification.
+
+The construction proceeds by recursively defining a sequence of coefficients as solutions to a
+polynomial equation in `k`. We must define these as generic polynomials using Witt vector API
+(`witt_vector.witt_mul`, `witt_polynomial`) to show that they satisfy the desired equation.
+
+This construction is described in Dupuis, Lewis, and Macbeth,
+[Formalized functional analysis via semilinear maps][dupuis-lewis-macbeth2022].
+We approximately follow an approach sketched on MathOverflow:
+<https://mathoverflow.net/questions/62468/about-frobenius-of-witt-vectors>
+
+The result is a dependency for the proof of `witt_vector.isocrystal_classification`,
+the classification of one-dimensional isocrystals over an algebraically closed field.
+-/
+
+noncomputable theory
+
+namespace witt_vector
+
+variables (p : ℕ) [hp : fact p.prime]
+variables {k : Type*} [field k]
+include hp
+local notation `𝕎` := witt_vector p
+
+namespace alg_closed_recursive
+
+/-!
+
+## The recursive case of the vector coefficients
+
+The first coefficient of our solution vector is easy to define below.
+In this section we focus on the recursive case.
+The goal is to turn `witt_poly_prod n` into a univariate polynomial
+whose variable represents the `n`th coefficient of `x` in `x * a`.
+
+For most of this section we work with terms of type `mv_polynomial (fin 2 × ℕ) ℤ`.
+We will eventually evaluate them in `k`, but first we must take care of a calculation
+that needs to happen in characteristic 0.
+-/
+
+open witt_vector finset
+open_locale big_operators
+
+section mv_poly
+open mv_polynomial
+
+omit hp
+
+/--
+```
+(∑ i in range n, (y.coeff i)^(p^(n-i)) * p^i.val) *
+  (∑ i in range n, (y.coeff i)^(p^(n-i)) * p^i.val)
+```
+-/
+def witt_poly_prod (n : ℕ) : mv_polynomial (fin 2 × ℕ) ℤ :=
+rename (prod.mk (0 : fin 2)) (witt_polynomial p ℤ n) *
+  rename (prod.mk (1 : fin 2)) (witt_polynomial p ℤ n)
+
+include hp
+
+lemma witt_poly_prod_vars (n : ℕ) :
+  (witt_poly_prod p n).vars ⊆ finset.univ.product (finset.range (n + 1)) :=
+begin
+  rw [witt_poly_prod],
+  apply subset.trans (vars_mul _ _),
+  apply union_subset;
+  { apply subset.trans (vars_rename _ _),
+    simp [witt_polynomial_vars,image_subset_iff] }
+end
+
+private lemma sum_ident_1 (n : ℕ) :
```

Comment:
I am really not qualified to judge, but I remember @fpvandoorn being [against](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/.E2.9C.94.20private.20and.20parameter/near/255174835) private declarations.